modules/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/TikaImpl.java

```diff
@@ -85,18 +93,40 @@ final class TikaImpl {
     /** singleton tika instance */
     private static final Tika TIKA_INSTANCE = new Tika(PARSER_INSTANCE.getDetector(), PARSER_INSTANCE);
 
+    /**
+     * parses PDFs
+     */
+    static String parsePDF(final InputStream content) throws IOException {
+        PDDocument pdfDocument;
+        try {
+            pdfDocument = PDDocument.load(content, MemoryUsageSetting.setupTempFileOnly());
```

Comment:
I know this is just a draft, but what if we did something like:
If the file is less than 50 MB, set the memory buffer to 3x the file size
If the file is greater than 50 MB, set the memory buffer to 150 MB
That way we'd handle it in memory when possible (which I assume is a lot faster), but keep it well below the 500 MB per PDF that we're currently allocating.

Reply:
it could be that they are processed in a serialized way

Reply:
I don't think that they are, but it's possible I'm missing something.